pkg/agent/controller/types.go

```diff
@@ -38,6 +39,8 @@ const (
 	portConflictReason = "ConflictingPorts"
 )
 
+var brokerResyncePeriod = time.Minute * 2
```

Comment:
Is this fixed duration or exponential backoff with max of 2 minutes? Do we log resync and at what severity?

Since user will need to create namespace for it to succeed, it may take long enough that it fills up the log.

If it is fixed at 2 minutes, then we need to consider it in E2E as it test will take up to 2 minutes to pass this requirement.

Reply:
> Is this fixed duration or exponential backoff with max of 2 minutes? 

Fixed. This isn't a retry mechanism but a periodic resync.

> Do we log resync and at what severity?
> 
> Since user will need to create namespace for it to succeed, it may take long enough that it fills up the log.
> 

Nothing is logged here by default.

> If it is fixed at 2 minutes, then we need to consider it in E2E as it test will take up to 2 minutes to pass this requirement.

I didn't write an E2E for this edge case for that reason. Ideally, we'd have to somehow reduce the duration for E2E.  I did recently add a test for when the namespace doesn't initially exist which is handled by the retry which essentially follows the same path in our code.